### PR TITLE
Production queue reorganization

### PR DESCRIPTION
**Pull request by Dilvish-fo**
This PR is for a  modest bit of cleanup and reorganization of ProductionQueue::Update() and some related code.  The primary intent is to better prepare this code for implementation of an Imperial Stockpile, for which I am also about to submit a PR, but the Imperial Stockpile may possibly linger for a while as a work in progress, whereas I think this reorganization is suitable to go ahead and put into master.

### CODE REVIEW

File: Empire/Empire.cpp [hunk 1]
```diff
@@ -179,9 +179,12 @@ namespace {
       * Elements will not receive funding if they cannot be produced by the
       * empire with the indicated \a empire_id this turn at their build location. */
     void SetProdQueueElementSpending(std::map<std::set<int>, float> available_pp,
-                                     const std::vector<std::set<int>>& queue_element_resource_sharing_object_groups,
-                                     ProductionQueue::QueueType& queue, std::map<std::set<int>, float>& allocated_pp,
-                                     int& projects_in_progress, int empire_id)
+                                     const std::vector<std::set<int> >& queue_element_resource_sharing_object_groups,
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: remove space between `> >`

File: Empire/Empire.cpp [hunk 2]
```diff
@@ -1005,35 +1011,33 @@ void ProductionQueue::Update() {
...
 
     const int TOO_MANY_TURNS = 500;     // stop counting turns to completion after this long, to prevent seemingly endless loops
-    const float TOO_LONG_TIME = 0.5f;   // max time in ms to spend simulating queue
+    const float TOO_LONG_TIME = 0.5f;   // max time in seconds to spend simulating queue
 
 
     // remove from simulated queue any items that can't be built due to not
-    // meeting their location conditions might be better to re-check
-    // buildability each turn, but this would require creating a simulated
+    // meeting their location conditions; can't really re-check
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: remove "really" or change to "feasibly"

File: Empire/Empire.cpp [hunk 3]
```diff
@@ -272,10 +260,17 @@ namespace {
...
-            std::tie(item_cost, build_turns) = queue_item_costs_and_times[key];
-            //DebugLogger() << "item " << queue_element.item.name << " costs " << item_cost << " for " << build_turns << " turns";
+            float item_cost = 1e6;  // dummy/default value, shouldn't ever really be needed
+            int build_turns = 1;    // dummy/default value, shouldn't ever really be needed
+            auto time_cost_it = queue_item_costs_and_times.find(key);
+            if (time_cost_it != queue_item_costs_and_times.end()) {
+                item_cost = time_cost_it->second.first;
+                build_turns = time_cost_it->second.second;
+            } else {
+                DebugLogger() << "item " << queue_element.item.name << " somehow failed time cost lookup for location "
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: not an error?

Comment (geoffthemedio) [reply Empire/Empire.cpp]: would move ` "somhow ..."` to the next line

File: Empire/Empire.cpp [hunk 4]
```diff
@@ -272,10 +260,17 @@ namespace {
...
+                              << location_id;
+            }
+                //DebugLogger() << "item " << queue_element.item.name << " costs " << item_cost << " for " << build_turns << " turns";
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: weird indenting or } placement

File: Empire/Empire.cpp [hunk 5]
```diff
@@ -969,12 +964,28 @@ void ProductionQueue::Update() {
...
+        elem.turns_left_to_completion = -1;
+    }
 
     // allocate pp to queue elements, returning updated available pp and updated
     // allocated pp for each group of resource sharing objects
-    SetProdQueueElementSpending(available_pp, queue_element_groups, m_queue,
-                                m_object_group_allocated_pp, m_projects_in_progress, m_empire_id);
-
+    SetProdQueueElementSpending(available_pp, queue_element_groups,
+                                queue_item_costs_and_times, is_producible,  m_queue,
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: odd extra space before `m_queue`

**Comment by geoffthemedio** [comment]
seems to run OK

File: Empire/Empire.cpp (hunk 3 above)
Comment (Dilvish-fo) [reply Empire/Empire.cpp]: I agree, that's probably better off being via the ErrorLogger

**Comment by geoffthemedio** [comment]
@Dilvish-fo  will you be attending to this some time soon?

**Comment by Dilvish-fo** [comment]
yes, I'll try to get it wrapped up in the next day or two

**Comment by geoffthemedio** [comment]
bump

**Comment by Dilvish-fo** [comment]
OK, cleaned up, and rebased.  I'll merge it now.

**Pull request merged**